vowpalwabbit/feature_group.h

```diff
@@ -346,36 +348,44 @@ struct features
 
     if (!space_names.empty())
     {
-      v_array<feature_slice> slice = v_init<feature_slice>();
+      std::vector<feature_slice> slice;
+      slice.reserve(indicies.size());
       for (size_t i = 0; i < indicies.size(); i++)
       {
-        feature_slice temp = {values[i], indicies[i] & parse_mask, *space_names[i].get()};
-        slice.push_back(temp);
+        slice.push_back({values[i], indicies[i] & parse_mask, *space_names[i].get()});
```

Comment:
nitpick: Its a lot better to have a constructor and use emplace_back

Reply:
I agree, but that's out of scope for now